pkg/steps/source.go

```diff
@@ -56,6 +57,18 @@ RUN git submodule update --init
 }
 
 func defaultPodLabels(jobSpec *api.JobSpec) map[string]string {
+	if refs := jobSpec.JobSpec.Refs; refs != nil {
```

Comment:
Why not use `jobSpec.ExtraRefs[0]` if it exists and no refs are set?

Reply:
addressed